### PR TITLE
Add onClose attribute to polaris-popover

### PR DESCRIPTION
**Pull request by tomnez**
Adds the [`onClose`](https://polaris.shopify.com/components/overlays/popover/onClose) property to the popover component

### CODE REVIEW

File: addon/components/polaris-popover.js [hunk 1]
```diff
@@ -87,10 +87,9 @@ export default Component.extend({
    *
    * @property onClose
    * @type {function(source: React.ReactElement)}
```
Comment (andrewpye) [inline addon/components/polaris-popover.js]: lol, we should probably update this to just be `function` 😉

File: addon/templates/components/polaris-popover.hbs [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {{#basic-dropdown
   verticalPosition=verticalPosition
   onOpen=(action "onOpen")
+  onClose=onClose
```
Comment (andrewpye) [inline addon/templates/components/polaris-popover.hbs]: Wrap this in an `action` helper as well to ensure context is always correctly maintained (it's what @vladucu would want 😉)

File: docs/popover.md [hunk 3]
```diff
@@ -6,7 +6,7 @@
 component](https://polaris.shopify.com/components/overlays/popover). This component uses [`ember-basic-dropdown`](https://github.com/cibernox/ember-basic-dropdown) to implement popover functionality. Note that the usage is slightly different from the React implementation so please pay attention to the examples below.
 
 
-**NOTE:** _the `active`, `activatorWrapper`, `preventAutofocus` and `onClose` properties are currently unimplemented._
+**NOTE:** _the `active`, `activatorWrapper`, and `preventAutofocus` properties are currently unimplemented._
```
Comment (andrewpye) [inline docs/popover.md]: Nice job remembering this 👍 Can you maybe add an example of using `onClose` here as well? Either work it into one of the existing examples or add a new one.

File: tests/integration/components/polaris-popover-test.js [hunk 4]
```diff
@@ -140,3 +140,35 @@ test('it renders the correct HTML with preferredPosition attribute', function(as
...
+
+      {{#popover.content}}
+        This is some sectioned popover content
+      {{/popover.content}}
+    {{/polaris-popover}}
+  `);
+
+  // open the popover
+  click(activatorSelector);
+
```
Comment (andrewpye) [inline tests/integration/components/polaris-popover-test.js]: Maybe check that the action hasn't run at this point just to be concrete about things? 🤷‍♂️

**Comment by andrewpye** [comment]
Also can you check what's up with Travis pls? Want a green light before merging this 👍

File: addon/templates/components/polaris-popover.hbs (hunk 2 above)
Comment (vladucu) [reply addon/templates/components/polaris-popover.hbs]: yes, please...it's also much easier to read 😄

**Comment by andrewpye** [comment]
@tomnez merge up from `master` - the mighty @vladucu has fixed the travis errors 🔥

**Comment by tomnez** [comment]
@andrewpye updated, and thanks to the mighty @vladucu for the travis fix 👊

**Pull request merged**